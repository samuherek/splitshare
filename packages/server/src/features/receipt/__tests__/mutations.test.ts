import gql from 'graphql-tag';
import createTestServer from '../../../test-utils/server';
import console = require('console');

describe('receipt mutations', () => {
  test('createReceipt: mutation should create and return a created receipt', async () => {
    const CREATE_RECEIPT = gql`
      mutation createReceipt($input: CreateReceiptInput!) {
        createReceipt(input: $input) {
          category
          comment
          createdAt
          currency
          id
          isSettlement
          paidAt
          title
          total
          updatedAt
        }
      }
    `;

    const { mutate } = await createTestServer({
      user: { id: 1 },
      models: {
        Receipt: {
          createOne: jest.fn(() => ({
            category: null,
            comment: null,
            createdAt: new Date('2000-01-01'),
            currency: 'EUR',
            id: 1,
            isSettlement: false,
            paidAt: new Date('2000-01-01'),
            title: 'first receipt',
            total: 20,
            updatedAt: new Date('2000-01-01'),
          })),
        },
      },
    });

    const res = await mutate({
      mutation: CREATE_RECEIPT,
      variables: {
        input: {
          billId: 1,
          title: 'first receipt',
          total: 20,
          currency: 'EUR',
          paidAt: '2000-01-01',
          paidById: 1,
          splits: [
            {
              userId: 1,
              value: 20,
            },
          ],
        },
      },
    });

    expect(res).toMatchSnapshot();
  });
});
