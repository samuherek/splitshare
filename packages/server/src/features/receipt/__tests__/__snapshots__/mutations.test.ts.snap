// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`receipt mutations createReceipt: mutation should create and return a created receipt 1`] = `
Object {
  "data": Object {
    "createReceipt": Object {
      "category": null,
      "comment": null,
      "createdAt": "2000-01-01",
      "currency": "EUR",
      "id": "1",
      "isSettlement": false,
      "paidAt": "2000-01-01",
      "title": "first receipt",
      "total": 20,
      "updatedAt": "2000-01-01T00:00:00.000Z",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
